test/models/system_variable_test.exs

```diff
@@ -39,6 +41,70 @@ defmodule CoursePlanner.SystemVariableTest do
       changeset = SystemVariable.changeset(%SystemVariable{}, %{@string_valid_attrs | value: ""})
       refute changeset.valid?
     end
+
+    test "changeset fails when input is empty but field is not required" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@string_valid_attrs | value: "", required: false})
+      assert changeset.valid?
+    end
+
+    test "changeset fails when input is more than 255 charactes" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@string_valid_attrs | value: String.duplicate("a", 500)})
+      refute changeset.valid?
+    end
+  end
+
+  describe "changeset for text input type validation" do
+    test "valid changeset when input is not empty" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@text_valid_attrs | value: "random"})
+      assert changeset.valid?
+    end
+
+    test "valid changeset when input can be interpreted as integer" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@text_valid_attrs | value: "112"})
+      assert changeset.valid?
+    end
+
+    test "valid changeset when input can be interpreted as boolean" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@text_valid_attrs | value: "true"})
+      assert changeset.valid?
+    end
+
+    test "changeset fails when input is empty" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@text_valid_attrs | value: ""})
+      refute changeset.valid?
+    end
+
+    test "valid changeset when input is more than 255 charactes" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@text_valid_attrs | value: String.duplicate("a", 500)})
+      assert changeset.valid?
+    end
+  end
+
+  describe "changeset for url input type validation" do
+    test "url is valid" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, @url_valid_attrs)
+      assert changeset.valid?
+    end
+
+    test "changeset fails when url has no scheme" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@url_valid_attrs | value: "www.sample.com"})
+      refute changeset.valid?
+    end
+
+    test "changeset fails when url has no host" do
+      changeset = SystemVariable.changeset(%SystemVariable{}, %{@url_valid_attrs | value: "http://"})
```

Comment:
can you change the error message from `Url should start with Http:// or Https://` to `Url should start with http:// or https://`?